### PR TITLE
Fix SizeBasedDocIdSet

### PR DESCRIPTION
**Pull request by Jackie-Jiang**
Fix the issue where the argument passed into SizeBasedDocIdSet is totalDocs instead of maxDocId

### CODE REVIEW

File: pinot-core/src/main/java/com/linkedin/pinot/core/operator/dociditerators/SizeBasedDocIdIterator.java [hunk 1]
```diff
@@ -18,38 +18,31 @@
 import com.linkedin.pinot.core.common.BlockDocIdIterator;
 import com.linkedin.pinot.core.common.Constants;
 
+
 public final class SizeBasedDocIdIterator implements BlockDocIdIterator {
```
Comment (kishoreg) [inline pinot-core/src/main/java/com/linkedin/pinot/core/operator/dociditerators/SizeBasedDocIdIterator.java]: add java docs explaining the behaviour, is it inclusive or exclusive

File: pinot-core/src/main/java/com/linkedin/pinot/core/operator/docidsets/SizeBasedDocIdSet.java [hunk 2]
```diff
@@ -39,12 +38,10 @@ public int getMaxDocId() {
 
   @Override
   public void setStartDocId(int startDocId) {
-    throw new UnsupportedOperationException();
```
Comment (kishoreg) [inline pinot-core/src/main/java/com/linkedin/pinot/core/operator/docidsets/SizeBasedDocIdSet.java]: why did you remove it, its ok for docIdSets to be immutable, i.e the min and max doc id cannot be changed later.

File: pinot-core/src/main/java/com/linkedin/pinot/core/operator/dociditerators/SizeBasedDocIdIterator.java (hunk 1 above)
Comment (Jackie-Jiang) [reply pinot-core/src/main/java/com/linkedin/pinot/core/operator/dociditerators/SizeBasedDocIdIterator.java]: Will add java doc into the BlockDocIdIterator for the behavior in another PR.

File: pinot-core/src/main/java/com/linkedin/pinot/core/operator/docidsets/SizeBasedDocIdSet.java (hunk 2 above)
Comment (Jackie-Jiang) [reply pinot-core/src/main/java/com/linkedin/pinot/core/operator/docidsets/SizeBasedDocIdSet.java]: They should not be changed, but calling it should not throw exception. That's why I leave it blank (no-op)

**Pull request merged**